src/plonkish/compiler/step_selector.rs

```diff
@@ -74,6 +74,26 @@ impl StepSelectorBuilder for SimpleStepSelectorBuilder {
             columns: Vec::new(),
         };
 
+        if unit.step_types.len() == 1 {
+            let step = unit.step_types.values().next().expect("step not found");
+
+            // use F::ONE for selector constantly on, F::ZERO for off
+            selector
+                .selector_expr
+                .insert(step.uuid(), PolyExpr::Const(F::ONE));
+
+            selector
+                .selector_expr_not
+                .insert(step.uuid(), PolyExpr::Const(F::ZERO));
+
+            selector
+                .selector_assignment
+                .insert(step.uuid(), vec![(PolyExpr::Const(F::ONE), F::ONE)]);
```

Comment:
@000wan this looks not the more readable way. I would make get_selector_assignment to return an Option, and then the code in assignment generator, could ignore this if None is returned.

Reply:
I agree with that. Now I modified `selector_assignment` to have an Option value. Could you check this again? @leolara